WebDriverAgentLib/Categories/XCElementSnapshot+FBHelpers.m

```diff
@@ -128,6 +132,30 @@ - (XCElementSnapshot *)fb_parentCellSnapshot
     }
     return targetCellSnapshot;
 }
+
+- (CGRect)fb_visibleFrameWithFallback
+{
+  CGRect thisVisibleFrame = [self visibleFrame];
+  if (!CGRectIsEmpty(thisVisibleFrame)) {
+    return thisVisibleFrame;
+  }
+  
+  NSDictionary *visibleFrameDict = (NSDictionary*)[self fb_attributeValue:@"XC_kAXXCAttributeVisibleFrame"];
+  if (visibleFrameDict == nil) {
+    return thisVisibleFrame;
+  }
+  
+  id x = [visibleFrameDict objectForKey:@"X"];
+  id y = [visibleFrameDict objectForKey:@"Y"];
+  id height = [visibleFrameDict objectForKey:@"Height"];
+  id width = [visibleFrameDict objectForKey:@"Width"];
+  if (x != nil && y != nil && height != nil && width != nil) {
```

Comment:
nit: this could be simplified to a single return:
```
return (x != nil && y != nil && height != nil && width != nil)
  ? CGRectMake([x doubleValue], [y doubleValue], [width doubleValue], [height doubleValue])
  : thisVisibleFrame;
```